activerecord/test/cases/relation/field_ordered_values_test.rb

```diff
@@ -13,6 +14,22 @@ def test_in_order_of
     assert_equal(order, posts.map(&:id))
   end
 
+  def test_in_order_of_with_enums_values
+    order    = %w[written published proposed]
+    books    = Book.in_order_of(:status, order.dup)
+    expected = order.map { |status| Array.new(Book.where(status: status).count, status) }.flatten
```

Comment:
Without looking at the book fixtures, I think it's a bit difficult to understand what the value of `expected` is.
The names of the book fixtures won't help much either.
```suggestion
    expected = [books(,adwr), books(:rdr), books(:ddd)]
```

Maybe creating records instead could be clearer? 🤷

Reply:
@p8 thank you very much for your review! 
I updated the tests to make them more easy to understand (by creating my own data as you suggested).